Compatibility with scipy 1.9 / test suite failure with cvxpy 1.2.2

Seen in https://github.com/conda-forge/cvxpy-feedstock/pull/72 on all platforms where we run tests:
```
=========================== short test summary info ============================
FAILED cvxpy/tests/test_grad.py::TestGrad::test_huber - ValueError: Inexact indices into sparse matrices are not allowed
FAILED cvxpy/tests/test_grad.py::TestGrad::test_partial_problem - ValueError: Inexact indices into sparse matrices are not allowed
===== 2 failed, 805 passed, 182 skipped, 552 warnings in 78.13s (0:01:18) ======
```

Based on the stack trace, this seems to be coming from scipy:
```
cvxpy/tests/test_grad.py:788: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[...]/lib/python3.9/site-packages/cvxpy/atoms/atom.py:402: in grad
    grad_self = self._grad(arg_values)
[...]/lib/python3.9/site-packages/cvxpy/atoms/elementwise/power.py:358: in _grad
    return [power.elemwise_grad_to_diag(grad_vals, rows, cols)]
[...]/lib/python3.9/site-packages/cvxpy/atoms/elementwise/elementwise.py:63: in elemwise_grad_to_diag
    return sp.dia_matrix((value, [0]), shape=(rows, cols)).tocsc()
[...]/lib/python3.9/site-packages/scipy/sparse/_dia.py:107: in __init__
    if isshape(arg1):
[...]/lib/python3.9/site-packages/scipy/sparse/_sputils.py:245: in isshape
    if isintlike(M) and isintlike(N):
```

Disallowing indexing with non-integers [enforced](https://github.com/scipy/scipy/blame/v1.9.3/scipy/sparse/_sputils.py#L228-L229) in scipy 1.9, as scipy has started following through on many longstanding open deprecations.

Since cvxpy [doesn't set](https://github.com/cvxpy/cvxpy/blame/v1.2.2/setup.py) an upper bound for scipy, this should be fixed soon (this will affect all other cvxpy releases as well...)

--- a/cvxpy/atoms/elementwise/elementwise.py
+++ b/cvxpy/atoms/elementwise/elementwise.py
@@ -60,7 +60,7 @@ class Elementwise(Atom):
         """
         if not np.isscalar(value):
             value = value.ravel(order='F')
-        return sp.dia_matrix((value, [0]), shape=(rows, cols)).tocsc()
+        return sp.dia_matrix((np.atleast_1d(value), [0]), shape=(rows, cols)).tocsc()
 
     @staticmethod
     def _promote(arg, shape: Tuple[int, ...]):

